lib/rules/disallow-function-prototype-extension.js

```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+var includes = require("../utils/includes");
+var extensionFixes = {
+  'property': "Use the computed('dependentKey', function() {}) syntax instead.",
+  'observes': "Use the observer('dependentKey', function() {}) syntax instead.",
+  'observesBefore': "Also, using before observers is deprecated."
```

Comment:
Agree on avoiding getting into deprecations. I just need to think of an appropriate message for `observesImmediately` and `observesBefore`. I'll probably say to use `observer` and `addObserver`.